doc/dox_comments/header_files/aes.h

```diff
@@ -1974,3 +1974,1696 @@ int wc_AesCtsDecryptUpdate(Aes* aes, byte* out, word32* outSz,
+    if (ret != 0) {
+        // failed to initialize
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesGcmInit
+    \sa wc_AesGcmDecryptUpdate
+*/
+int wc_AesGcmDecryptInit(Aes* aes, const byte* key, word32 len,
+                         const byte* iv, word32 ivSz);
+
+/*!
+    \ingroup AES
+    \brief This function performs an update step of AES GCM decryption.
+    It processes ciphertext and/or additional authentication data (AAD)
+    in a streaming fashion.
+
+    \return 0 On success.
+    \return BAD_FUNC_ARG If aes is NULL, or if parameters are invalid.
+
+    \param aes pointer to the AES structure
+    \param out pointer to buffer to store plaintext (can be NULL if sz=0)
+    \param in pointer to ciphertext to decrypt (can be NULL if sz=0)
+    \param sz length of ciphertext in bytes
+    \param authIn pointer to additional authentication data (can be NULL)
+    \param authInSz length of AAD in bytes
+
+    _Example_
+    \code
+    Aes aes;
+    byte key[16] = { }; // 128-bit key
+    byte iv[12] = { }; // nonce
+    byte ciphertext[100] = { }; // encrypted data
+    byte plaintext[100];
+    byte aad[20] = { }; // additional data
+
+    wc_AesInit(&aes, NULL, INVALID_DEVID);
+    wc_AesGcmDecryptInit(&aes, key, 16, iv, 12);
+    int ret = wc_AesGcmDecryptUpdate(&aes, plaintext, ciphertext, 100,
+                                     aad, 20);
+    if (ret != 0) {
+        // decryption failed
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesGcmDecryptInit
+    \sa wc_AesGcmDecryptFinal
+*/
+int wc_AesGcmDecryptUpdate(Aes* aes, byte* out, const byte* in, word32 sz,
+                           const byte* authIn, word32 authInSz);
+
+/*!
+    \ingroup AES
+    \brief This function finalizes AES GCM decryption and verifies the
+    authentication tag. This must be called after all data has been
+    processed with wc_AesGcmDecryptUpdate.
+
+    \return 0 On success.
+    \return AES_GCM_AUTH_E If authentication tag verification fails.
+    \return BAD_FUNC_ARG If aes or authTag is NULL, or if authTagSz is
+    invalid.
+
+    \param aes pointer to the AES structure
+    \param authTag pointer to the authentication tag to verify
+    \param authTagSz length of the authentication tag in bytes
+
+    _Example_
+    \code
+    Aes aes;
+    byte key[16] = { }; // 128-bit key
+    byte iv[12] = { }; // nonce
+    byte ciphertext[100] = { }; // encrypted data
+    byte plaintext[100];
+    byte authTag[16] = { }; // received tag
+
+    wc_AesInit(&aes, NULL, INVALID_DEVID);
+    wc_AesGcmDecryptInit(&aes, key, 16, iv, 12);
+    wc_AesGcmDecryptUpdate(&aes, plaintext, ciphertext, 100, NULL, 0);
+    int ret = wc_AesGcmDecryptFinal(&aes, authTag, 16);
+    if (ret != 0) {
+        // authentication failed
+    }
+    wc_AesFree(&aes);
+    \endcode
+
+    \sa wc_AesGcmDecryptUpdate
+    \sa wc_AesGcmEncryptFinal
+*/
+int wc_AesGcmDecryptFinal(Aes* aes, const byte* authTag, word32 authTagSz);
```

Comment:
`wc_AesGcmDecryptFinal` and most of these wc_AesGcm functions are already commented in the source. Not sure if want duplication.